csrc/scheduler/multi_matmul.cpp

```diff
@@ -1296,22 +1294,292 @@ class MultipleMatmulScheduler {
+        //!  enabled for matmul scheduler.
+        parallel_types = allParallelTypesExcept({ParallelType::Vectorize});
+      }
+      scheduler_utils::parallelizeAllLike(
+          output_d, -1, cached_tvs, parallel_types);
+
+      // The cached EPILOGUE_INPUT tvs are not needed anymore
+      cached_tvs.clear();
+    }
   }
 
   void scheduleSplitKSum() {
-    NVF_THROW("scheduleSplitKSum is not yet implemented");
+    if (params_->splitk_factor == 1) {
+      return;
+    }
+    for (TensorView* splitk_sum : splitk_sums_) {
+      // Always use serial grid reduction for split-K sum
+      splitk_sum->definition()->as<ReductionOp>()->requestSerialGridReduction();
+
+      if (params_->use_smem_epilogue) {
+        // Now that transforms are propagated backward to smem_epilogue, which
+        // is before splitk_sum, we can vectorize the inner-most non-trivial
+        // dimension of splitk_sum
+        //
+        // Note that the split-K reduction is the inner-most dimension.
+        Val* vec_ext = splitk_sum->axis(-2)->extent();
+        NVF_ERROR(vec_ext->isConstInt());
+        int64_t vec_ext_int = vec_ext->evaluate().as<int64_t>();
+        splitk_sum->axis(-1)->parallelize(ParallelType::BIDz);
+        splitk_sum->axis(-3)->parallelize(ParallelType::TIDx);
+        if (vec_ext_int * dataTypeSize(splitk_sum->dtype()) > 16) {
+          // NOTE: We might encounter an illegal vectorization size if we are
+          // using Float for this reduction and Half for output. So here we
+          // first check whether the vectorize size is at most 16 bytes. If not,
+          // then we split into an unrolled loop that will do multiple
+          // vectorized reads/writes instead. Note that we reorder such that the
+          // axes are in order UR TIDx V.
+          splitk_sum->split(
+              -2, 16 / dataTypeSize(splitk_sum->dtype()), /*inner_split=*/true);
+          splitk_sum->axis(-3)->parallelize(ParallelType::Unroll);
+          splitk_sum->reorder({{-4, -3}});
+          // In this case, we have [... iUR iTx rBz iS]
+        }
+        splitk_sum->reorder({{-2, -1}});
+      } else { // no smem epilogue
+        // Reorder to place the split-K reduction next to innermost [... rBz iS]
+        splitk_sum->reorder({{-9, -2}});
+      }
+      // Vectorize inner-most dimension [... (iUR iTx) rBz iV]
+      splitk_sum->axis(-1)->parallelize(ParallelType::Vectorize);
+    }
   }
 
   void setUpInlining() {
-    NVF_THROW("setUpInlining is not yet implemented");
+    // auto inline for all tensors except register tensors
+    std::unordered_set<TensorView*> smem_loads_and_mma_inputs;
+    smem_loads_and_mma_inputs.insert(acrs_.begin(), acrs_.end());
+    smem_loads_and_mma_inputs.insert(bcrs_.begin(), bcrs_.end());
+    smem_loads_and_mma_inputs.insert(abs_.begin(), abs_.end());
+    smem_loads_and_mma_inputs.insert(bbs_.begin(), bbs_.end());
+    inlineMost(ir_utils::allTvsExcept(fusion_, smem_loads_and_mma_inputs));
```

Comment:
This was originally
```c++
inlineMost(ir_utils::allTvsExcept(fusion, {acr, bcr, ab, bb}));
```
Now I build the `except` argument ahead of time since we no longer have a fixed number of tensors.